### ISSUE TITLE
Reconsider mouse support for non-hyperlink file paths

### ISSUE THREAD

**Issue by sirthias** [kovidgoyal/kitty#6909]
I'm coming from iTerm2 on macOS and really like kitty's approach to customization.
It's much more flexible and hackable than iTerm2, with one (for my use case) serious caveat:
While kitty's support for hyperlinks is great, most tools, compilers, linters, formatters, etc. **don't** output hyperlinks but simple file system paths. Some of them can be coaxed into producing hyperlinks, but pushing the requirement to produce hyperlinks into a million tools rather than trying to deal with file paths on the terminal level, seems like the wrong approach to me.

In #977 it was argued than file paths are hard to detect.
Yet the hints kitty already does it nicely and robustness of detection is rarely a problem.

The only thing missing from the "hints" solution is mouse support.

In iTerm2 it's easy to have a custom script run when I ctrl-click on any file system path (or hyperlink) plus -- crucially -- line and col number fragment. My script then decides which editor to open based on a number of clues.

Currently, in kitty it appears to be simply impossible to mimic this behavior, which is why I currently cannot switch to kitty as my default terminal even though almost everything else is much better than iTerm2.

Please reconsider your stance that keyboard-only hints a good enough, because -- unfortunately -- for quite a few people they aren't.

**Comment by sirthias** [comment]
One more comment:
iTerm2 verifies that a path "candidate" refers to an existing file before accepting it as a _true_ file path, which works very well in practice. In the 10+ years that I've been using it quite heavily I never had a problem with false positives/negatives.

**Comment by sirthias** [comment]
Fair enough.
How about a facility that would allow me to transform the currently visible window content with some custom logic when I press CTRL or some other modifier?
Then I could hook in my "detect file" paths logic myself and transform them into hyperlinks on the terminal level. Afterward the existing hyperlinking logic would be there to do its magic...

**Comment by kovidgoyal** [comment]
You can already pass the screen contents to arbitrary programs/scripts and have
their output displayed in an overlay window, that's how the hint kitten
works. You can write your own to do whatever you like.
See https://sw.kovidgoyal.net/kitty/launch/

**Comment by sirthias** [comment]
Yes, that's nice and indeed very powerful. It could probably launch the space shuttle.
However, I don't see a way to use it for quickly opening `/path/file.ext:1:2`, which some previous command has output in a multi-page stream of error messages, in my IDE via the mouse.

Maybe I could open an overlay window exactly over the currently active one, fill it will the screen contents with all paths replaced by hyperlinks and automatically close it again after the user has clicked on one of the links.
It could work but would be a step back from the simplicity of the solution I have with my current setup.

In any case, thank you for making kitty!
Its a great addition to the terminal space, even without full-blown mouse support.

**Comment by kovidgoyal** [comment]
The only difference with your current workflow would be the need to press a key befor clicking with the mouse.

**Comment by sirthias** [comment]
That wouldn't be the problem.
But opening an overlay window exactly at the right position, with the right dimensions, so that it overlays perfectly with the current one, then transforming and showing the contents, with all colors and decorations, and then closing it again after the click, appears to me like quite a bit of overhead.
And it would take me a considerable amount of time to implement, if the API even allows for all of this to be done from outside kitty.

Thank you again!

**Comment by kovidgoyal** [comment]
overlays open that way automatically you dont need to do anything other
than write a script with a few regexps to create the hyperlinks for
STDIN and print the result to STDOUT and then wait for a key press. That
alone will get you to a point where you have to 

press a key + click + press another key to exit

If you want to eliminate the key press to exit, you would need to write a
small ~ 500 line command line program that accepts mouse clicks and
runs whatever program you like based on whatever is clicked and then
exits. There exist many TUI libraries in many programming languages that
make developing this easy.

There is no kitty specific API involved at all in any of this, beyond
one line in kitty.conf with the mapping for the launch command to invoke
your program and pass it the screen contents over STDIN.

All that said, I would advise you to just use the hints kitten once you
start you will find it far more efficient than using the mouse.

**Comment by sirthias** [comment]
Hmmm... ok.
I'll give the keyboard-only solution via the hints a try and see how it feels...

Thank you again so much for having taken the time to convince me, somewhat! :)
